app/shopcart.py

```diff
@@ -411,6 +557,73 @@ def create_products(sid):
 ######################################################################
 @app.route('/shopcarts/<int:sid>/products/<int:sku>', methods=['PUT'])
 def put_product(sid, sku):
+
+    """
+    Updates a Product
+    This endpoint will update a Product based the data in the body that is posted
+    ---
+    tags:
+      - Shopcarts
+    consumes:
+      - application/json
+    produces:
+      - application/json
+    parameters:
+      - name: sid
+        in: path
+        description: the sid of the shopcart whose product you want to update
+        required: true
+        type: integer
+      - name: sku
+        in: path
+        description: the sku of the product you are looking for
+        required: true
+        type: integer
+      - in: body
+        name: body
+        required: true
+        schema:
+          id: data
+          required:
+            - sku
+            - name
+            - quantity
+            - unitprice
+          properties:
+            sku:
+                type: integer
+                description: unique id of the particular product
+            name:
+                type: string
+                description: name of the particular product
+            quantity:
+                type: integer
+                description: quantity of the particular product
+            unitprice:
+                type: number
+                description: price of the particular product
+    responses:
+      200:
+        description: Product updated
+        schema:
```

Comment:
If we are specifying the sku of the product to update then there should be no need to send the whole list in the body of the request.